code/game/objects/items/toys.dm

```diff
@@ -1532,3 +1533,35 @@ GLOBAL_LIST_EMPTY(intento_players)
 #undef TIME_TO_BEGIN
 #undef TIME_PER_DEMO_STEP
 #undef TIME_TO_RESET_ICON
+
+/*
+ * Groan Tube (thing that goes UUAAAA AAAUUU)
+ */
+/obj/item/toy/groan_tube
+	name = "groan tube"
+	desc = "UUAAAA...   AAAUUUU"
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "groan_tube"
+	verb_say = "groans"
+	var/cooldown = FALSE
+	var/cooldown_time = 20
+	var/flipped = FALSE
+
+/obj/item/toy/groan_tube/attack_self(mob/user)
+	if(!cooldown)
+
+		to_chat(user, span_notice("You flip \the [src]."))
+		flick("groan_tube_flip", src)
+		if(flipped)
+			playsound(loc, 'sound/items/aaau.ogg', 50, FALSE, 3)
+			say("AAAUUU")
+		else
+			playsound(loc, 'sound/items/uuua.ogg', 50, FALSE, 3)
+			say("UUAAAA")
+
+		flipped = !flipped
+		cooldown = TRUE
+		addtimer(VARSET_CALLBACK(src, cooldown, FALSE), cooldown_time)
```

Comment:
cute use but this is slower than just using the COOLDOWN macros.

COOLDOWN_DECLARE()
COOLDOWN_START()
ON_COOLDOWN()

Reply:
ok does that